### PR TITLE
Prevent GraphArg from keeping real tensors live

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #100515

This may potentially fix an OOM on IG Cover model (Meta only: T152238176)

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

### CODE REVIEW

File: torch/_dynamo/variables/builder.py [hunk 1]
```diff
@@ -135,8 +136,16 @@ class GraphArg:
     # or not.
     is_tensor: bool = True
 
+    @property
+    def example(self):
+        if isinstance(self._example, TensorWeakRef):
+            return self._example()
+        else:
+            return self._example
```
Comment (voznesenskym) [inline torch/_dynamo/variables/builder.py:144]: assert SymInt so that real tensors don't creep back in?

**Review by voznesenskym — APPROVED** [review]

**Review by yf225 — APPROVED** [review]
Can confirm that this fixes the OOM error. Thanks!

File: torch/_dynamo/variables/builder.py (hunk 1 above)
Comment (ezyang) [reply torch/_dynamo/variables/builder.py:143]: I'd probably prefer to unwind the example/fake_example Tensor/SymInt/FakeTensor confusion

**Pull request closed without merging**